db/dcrpg/pgblockchain.go

```diff
@@ -2906,9 +2912,73 @@ func (pgb *ChainDB) PurgeBestBlocks(N int64) (*dbtypes.DeletionSummary, int64, e
 
 	summary := dbtypes.DeletionSummarySlice(res).Reduce()
 
+	// Rewind stake database to this height.
+	var stakeDBHeight int64
+	stakeDBHeight, err = pgb.RewindStakeDB(pgb.ctx, height, true)
+	if err != nil {
+		return nil, height, pgb.replaceCancelError(err)
+	}
+	if stakeDBHeight != height {
+		err = fmt.Errorf("rewind of StakeDatabase to height %d failed, "+
+			"reaching height %d instead", height, stakeDBHeight)
+		return nil, height, pgb.replaceCancelError(err)
+	}
+
 	return &summary, height, err
 }
 
+// RewindStakeDB attempts to disconnect blocks from the stake database to reach
+// the specified height. A channel must be provided for signaling if the rewind
```

Comment:
The part about a channel for signalling can either be removed or changed to something like "A Context may be provided to allow cancellation of the rewind process."